unit/analyses/call_graph.cpp

```diff
@@ -0,0 +1,113 @@
+
+#include <iostream>
+
+#include <catch.hpp>
+
+#include <analyses/call_graph.h>
+
+#include <util/symbol_table.h>
+#include <util/std_code.h>
+
+#include <goto-programs/goto_convert_functions.h>
+
+SCENARIO("call_graph",
+  "[core][util][call_graph]")
+{
+
+  symbol_tablet symbol_table;
+
+  code_typet void_function_type;
+
+  {
```

Comment:
To reduce number of lines - you might consider pulling out a couple of utility functions like:

```C++
symbolt create_void_function_symbol(const irep_idt &name, const codet &code)
codet create_two_calls(const irep_idt &call1, const irep_idt &call2)
```

I don't think they are useful outside of this test so I wouldn't bother pulling them into a utility file. 

I think it might make the tests clearer if you provide a simple comment that shows the generated code:

```C++
void A() {
  A();
  B();
}
void B()
{
  C();
  D();
}
void C() { }
void D() { }
```